File: js/ejer60.js

const candidates = {
  1: 'Paloma Valencia',
  2: 'Abelardo de la Espriella',
  3: 'Ivan Cepeda'
};

const counts = {1: 0, 2: 0, 3: 0};

function preguntarEntero(text) {
  const entrada = prompt(text);
  if (entrada === null) return null; // cancelado
  const n = Number(entrada.trim());
  if (Number.isInteger(n)) return n;
  return NaN;
}

alert('Sistema de votación: elige un candidato por número. Se mostrarán resultados al final.');
let votantes = preguntarEntero('¿Cuántos votantes hay?');
if (votantes === null) {
  console.log('Operación cancelada por el usuario.');
} else {
  while (!Number.isInteger(votantes) || votantes <= 0) {
    votantes = preguntarEntero('Ingresa un número válido de votantes:');
    if (votantes === null) {
      console.log('Operación cancelada por el usuario.');
      break;
    }
  }

  if (votantes !== null) {
    for (let i = 1; i <= votantes; i++) {
      let opcion = prompt(`Votante ${i}:\n1. ${candidates[1]}\n2. ${candidates[2]}\n3. ${candidates[3]}\nIngresa 1, 2 o 3 para votar (o deja vacío para abstenerse):`);
      if (opcion === null) {
        console.log('Usuario canceló durante la votación.');
        break;
      }
      opcion = opcion.trim();
      if (opcion === '') {
        console.log(`Votante ${i}: abstención.`);
        continue;
      }
      const num = Number(opcion);
      if (![1,2,3].includes(num)) {
        alert('Opción inválida. Se volverá a pedir el voto.');
        i--;
        continue;
      }
      counts[num]++;
      console.log(`Votante ${i}: voto para ${candidates[num]}`);
    }

    // Mostrar resultados
    console.log('\nResultados:');
    for (const k of [1,2,3]) {
      console.log(`${candidates[k]}: ${counts[k]} voto(s)`);
    }

    const maxVotes = Math.max(counts[1], counts[2], counts[3]);
    const winners = Object.keys(counts).filter(k => counts[k] === maxVotes && maxVotes > 0);

    if (maxVotes === 0) {
      alert('No se registraron votos.');
    } else if (winners.length === 1) {
      alert(`Ganador: ${candidates[winners[0]]} con ${maxVotes} voto(s).`);
    } else {
      const names = winners.map(k => candidates[k]).join(', ');
      alert(`Empate entre: ${names} con ${maxVotes} voto(s) cada uno.`);
    }
  }
}

File: python/ejer60.py

candidates = {
    "1": "Paloma Valencia",
    "2": "Abelardo de la Espriella",
    "3": "Ivan Cepeda"
}

counts = {key: 0 for key in candidates}

def pedir_entero(prompt_text):
    while True:
        try:
            v = input(prompt_text).strip()
            if v == "":
                print("Entrada vacía. Intenta de nuevo.")
                continue
            n = int(v)
            return n
        except ValueError:
            print("Por favor ingresa un número entero.")

print("Sistema de votación: elige un candidato por número.")
print("Opciones:")
for k, name in candidates.items():
    print(f"  {k}. {name}")

votantes = pedir_entero("¿Cuántos votantes hay? ")

for i in range(1, votantes + 1):
    print(f"\nVotante {i}:")
    while True:
        opcion = input("Ingresa 1, 2 o 3 para votar (o 'x' para abstenerse): ").strip()
        if opcion.lower() == 'x':
            print("Abstención registrada.")
            break
        if opcion in candidates:
            counts[opcion] += 1
            print(f"Voto registrado para {candidates[opcion]}")
            break
        print("Opción inválida. Intenta de nuevo.")

print("\nResultados:")
for k, name in candidates.items():
    print(f"  {name}: {counts[k]} voto(s)")

max_votes = max(counts.values())
winners = [candidates[k] for k, v in counts.items() if v == max_votes]

if max_votes == 0:
    print("No se registraron votos.")
elif len(winners) == 1:
    print(f"\nGanador: {winners[0]} con {max_votes} voto(s).")
else:
    print("\nEmpate entre:")
    for w in winners:
        print(f" - {w}")
    print(f"con {max_votes} voto(s) cada uno.")
